lib/rules/no-irregular-whitespace.js

```diff
@@ -47,12 +48,52 @@ module.exports = function(context) {
     function removeInvalidNodeErrors(node) {
         if (typeof node.value === "string") {
             // If we have irregular characters remove them from the errors list
-            if (node.value.match(irregularWhitespace)) {
+            if (node.raw.match(irregularWhitespace) || node.raw.match(irregularLineTerminators)) {
                 removeStringError(node);
             }
         }
     }
 
+    function checkForIrregularWhitespace(node) {
+        var sourceLines = context.getSourceLines();
+
+        sourceLines.forEach(function (sourceLine, lineIndex) {
+            var lineNumber = lineIndex + 1,
+                location,
+                match;
+
+            while ((match = irregularWhitespace.exec(sourceLine)) !== null) {
```

Comment:
Drive-by-fix: using a while instead of if, allows to detect multiple errors in a single line.